documentation/gnmi/design/sonic-dash.yang

```diff
@@ -0,0 +1,490 @@
+
+                leaf terminating {
+                    type boolean
+                    default false;
+                    description "If set to true, stop processing further rules";
+                }
+
+                leaf-list ip_protocol {
+                    description "IP Protocol (tcp or udp or icmp etc)";
+                    type stypes:ip-protocol-type;
+                }   
+
+                leaf-list src_addr {
+                     type union {
+                         type stypes:sonic-ip4-prefix;
+                         type stypes:sonic-ip6-prefix;
+                     }
+                }
+
+                leaf-list dst_addr {
+                     type union {
+                         type stypes:sonic-ip4-prefix;
+                         type stypes:sonic-ip6-prefix;
+                     }
+                }
+
+                leaf-list src_port {
+                     description "List of L4 source port range as pattern '0-65365'"
+                     type string {
+                         pattern '([0-9]{1,4}|[0-5][0-9]{4}|[6][0-4][0-9]{3}|[6][5][0-2][0-9]{2}|[6][5][3][0-5]{2}|[6][5][3][6][0-5])-([0-9]{1,4}|[0-5][0-9]{4}|[6][0-4][0-9]{3}|[6][5][0-2][0-9]{2}|[6][5][3][0-5]{2}|[6][5][3][6][0-5])';
+                     }
+                }
+
+                leaf-list dst_port {
+                     description "List of L4 dst port range as pattern '0-65365'"
+                     type string {
+                         pattern '([0-9]{1,4}|[0-5][0-9]{4}|[6][0-4][0-9]{3}|[6][5][0-2][0-9]{2}|[6][5][3][0-5]{2}|[6][5][3][6][0-5])-([0-9]{1,4}|[0-5][0-9]{4}|[6][0-4][0-9]{3}|[6][5][0-2][0-9]{2}|[6][5][3][0-5]{2}|[6][5][3][6][0-5])';
+                     }
+                }
+
+            } /* end of list DASH_ACL_RULE_LIST */
+        } /* end of container DASH_ACL_RULE */
+
+        container DASH_APPLIANCE {
+            description "DASH general appliance configs; part of APP_DB";            
+
+            list DASH_APPLIANCE_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf sip {
+                     type inet:ip-address;
+                }
+
+                leaf-list vni {
+                    type  uint32 {
+                        range 0..16777215;
+                    }
+                }
+
+            } /* end of list DASH_APPLIANCE_LIST */
+        } /* end of container DASH_APPLIANCE */
+
+        container DASH_ROUTING_TYPE {
+            description "DASH Routing types and actions; part of APP_DB";            
+
+            list DASH_ROUTING_TYPE_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern "direct|vnet|vnet-direct|drop|appliance|privatelink|privatelinknsg|servicetunnel";
+                    }
+                }
+
+                list actions {
```

Comment:
Do we need to restrict no. of action entries per routing type? e.g direct routing type can have max. 4 actions.